server/pbench/bin/pbench-sync-satellite

```diff
@@ -136,53 +160,85 @@ for host in $hosts ;do
+    processed=$(wc -l < $logdir/$host/md5-checks.log)
+    nprocessed=$((nprocessed + processed))
     grep 'OK' $logdir/$host/md5-checks.log > $logdir/$host/ok-checks.log
     if [ -s $logdir/$host/ok-checks.log ] ;then
         md5_pass=$(cat $logdir/$host/ok-checks.log | sed -n 's/: OK//p')
         for x in $md5_pass; do
-            ln -s $x SATELLITE-MD5-PASSED/$x
+            ln -sf $PWD/$x SATELLITE-MD5-PASSED/$x
+            status=$?
+            if [[ $status != 0 ]] ;then
+                echo "FAILED: ln -sf $PWD/$x SATELLITE-MD5-PASSED/$x"
+                exit 1
+            fi
         done
     fi
     grep 'FAIL' $logdir/$host/md5-checks.log > $logdir/$host/fail-checks.log
+    cat $logdir/$host/fail-checks.log >> $tmp/mail_content.log
+    failed_md5=$(wc -l < $logdir/$host/fail-checks.log)
+    nfailed_md5=$((nfailed_md5 + failed_md5))
     if [ -s $logdir/$host/fail-checks.log ] ;then
         md5_fail=$(cat $logdir/$host/fail-checks.log | sed -n 's/: FAIL//p')
         for x in $md5_fail; do
-            ln -s $x SATELLITE-MD5-FAILED/$x
+            ln -sf $PWD/$x SATELLITE-MD5-FAILED/$x
+            status=$?
+            if [[ $status != 0 ]] ;then
+                echo "FAILED: ln -sf $PWD/$x SATELLITE-MD5-FAILED/$x"
+                exit 1
+            fi
         done
     fi
     # create symlinks for the synced tarballs in TODO
     if [ -s $logdir/$host/ok-checks.log ] ;then
         for x in $md5_pass; do
-            ln -s $x TODO/$x
+            ln -sf $PWD/$x TODO/$x
+            status=$?
+            if [[ $status != 0 ]] ;then
+                echo "FAILED: ln -sf $PWD/$x TODO/$x"
+                exit 1
+            fi
         done
     fi
     # save the contents of ok checks to further use it for state change
     if [ -s $logdir/$host/ok-checks.log ] ;then
         state_list=$(cat $logdir/$host/ok-checks.log | sed -n 's/: OK//p')
         for x in $state_list; do
-            echo "$host/TO-SYNC/$x" >> $tmp/change_state.log
+            echo "$host/TO-SYNC/$x" >> $LOGSDIR/change_state.$prefix.log
         done
     fi 
     popd > /dev/null
 done
 
 # change the state of the tarballs on remote
-ssh $remotehost "cd $remotearchive; /opt/pbench-server/bin/pbench-satellite-state-change" < $tmp/change_state.log > $logdir/$host/mv.log 2>&1
-
-nprocessed=$(wc -l < $logdir/$host/md5-checks.log)
-nfailed_md5=$(wc -l < $logdir/$host/fail-checks.log)
+if [ -s $LOGSDIR/change_state.$prefix.log ] ;then
+    ssh $remotehost "cd $remotearchive; /opt/pbench-server/bin/pbench-satellite-state-change" < $LOGSDIR/change_state.$prefix.log > $logdir/mv.log 2>&1
+    status=$?
+    if [[ $status != 0 ]] ;then
+        echo "FAILED: ssh $remotehost "cd $remotearchive; /opt/pbench-server/bin/pbench-satellite-state-change" < $LOGSDIR/change_state.$prefix.log"
+        echo "Unable to change the state in $remotehost" | mailx -s \
+        "FAILED: $PROG: $prefix: ssh to $remotehost failed"\
+        $mail_recipients
```

Comment:
You should test this (e.g. by deliberately setting the ssh status to 1) and directing the email to yourself. Pretend that you don't know anything about this script and that you get the mail message out of the blue: would you be able to understand what went wrong and fix it just by reading the mail message?